File: frontend/src/types/recipe.ts
import { Ingredient } from "./ingredient"
import { instruction } from "./instruction";
import { SearchItem } from "./searchItem";
import { Step } from "./step";

export interface recipeCommon {
    "id"?: string,
    "title": string,
    "imageUrls"?: string[],
    "imageType"?: string,
    "servings": number,
    "readyInMinutes"?: number,
    "cookingMinutes"?: number,
    "preparationMinutes"?: number,
    "sourceName": string,
    "sourceUrl"?: string,
    "cuisine"?: string,
    "instructions": instruction[],
    "diet"?: string,
    "type"?: string,
    "extendedIngredients": Ingredient[],
    "calories"?: number,
    "protein"?: number,
    "carbohydrate"?: number,
    "fat"?: number,
    "summary"?: string
}

export interface recipeSearchRes {
    results: SearchItem[],
    offset: number,
    number: number,
    totalResults: number
}

export interface RecipeDraft {
    title: string,
    servings: number,
    preparationMinutes?: number,
    type?: string, 
    cuisine?: string,
    diet?: string,
    prepTime?: number,
    ingredients: Ingredient[],
    steps: Step[],
    sourceName?: string,
    images?: (File | string)[]
}

export enum Tab {
    UPLOAD = 'uploadRecipes',
    OVERVIEW = 'overview',
    INGREDIENTS = 'ingredients',
    STEPS = 'steps',
    NOTES = 'notes',
    IMAGES = 'images'
}

File: backend/API/Entity/Recipe.cs
namespace API.Entity
{
    public class Recipe
    {
        public long Id { get; set; }
        public string Title { get; set; }
        public Dictionary<string, string> ImageInfo { get; set; }
        public int Servings { get; set; }
        public int PreparationMinutes { get; set; }
        public int CookingMinutes { get; set; }
        public string SourceName { get; set; }
        public string SourceUrl { get; set; }
        public string Cuisine { get; set; }
        public List<string> Diets { get; set; }
        public List<Instruction> Instructions { get; set; }
        public string DishType { get; set; }
        public string Summary { get; set; }
        public decimal Calories { get; set; }
        public decimal Protein { get; set; }
        public decimal Carbohydrate { get; set; }
        public decimal Fat { get; set; }
        public List<RecipeIngredient> RecipeIngredients { get; set; } = [];

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public string UserId { get; set; }
    }
}
